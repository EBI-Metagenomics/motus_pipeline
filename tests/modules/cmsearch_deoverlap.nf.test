nextflow_process {

    name "Test module cmsearch-deoverlap"
    script "modules/cmsearch_deoverlap.nf"
    process "CMSEARCH_DEOVERLAP"

    test("cmsearch-deoverlap") {

        when {
            params {
                matches = "${baseDir}/tests/modules/fixtures/cmsearch/output/mgtest_merged.fastq.cmsearch_matches.tbl"
                clan = "${baseDir}/tests/modules/fixtures/cmsearch/ribo.claninfo"
            }
            process {
                """
                input[0] = channel.fromPath("${params.clan}", checkIfExists: true)
                input[1] = channel.fromPath("${params.matches}", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}