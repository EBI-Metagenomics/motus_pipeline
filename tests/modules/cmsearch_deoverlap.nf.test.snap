{
    "cmsearch": {
        "content": [
            {
                "cmsearch_deoverlap": [
                    "mgtest_merged.fastq.cmsearch_matches.tbl.deoverlapped:md5,bc0275dfe8459658458b7fe6776fa9ba"
                ]
            }
        ],
        "timestamp": "2023-03-14T14:21:14+0000"
    },
    "cmsearch-deoverlap": {
        "content": [
            {
                "cmsearch_deoverlap": [
                    "mgtest_merged.fastq.cmsearch_matches.tbl.deoverlapped:md5,bc0275dfe8459658458b7fe6776fa9ba"
                ]
            }
        ],
        "timestamp": "2023-03-14T14:21:14+0000"
    }
}